website/docs/r/database_secret_backend_role.md

```diff
@@ -78,6 +78,40 @@ The following arguments are supported:
 * `max_ttl` - (Optional) The maximum number of seconds for leases for this
   role.
 
+* `credential_type` (Optional) – Specifies the type of credential that
+  will be generated for the role. Options include: `password`, `rsa_private_key`, `client_certificate`.
+  See the plugin's API page for credential types supported by individual databases.
+
+* `credential_config` (Optional) – Specifies the configuration
+  for the given `credential_type`.
+
+  The following options are available for each `credential_type` value:
+
+  * `password`
+    * `password_policy` (Optional) - The [policy](/vault/docs/concepts/password-policies)
+      used for password generation. If not provided, defaults to the password policy of the
+      database [configuration](/vault/api-docs/secret/databases#password_policy).
+
+  * `rsa_private_key`
+    * `key_bits` (Optional) - The bit size of the RSA key to generate. Options include:
+      `2048`, `3072`, `4096`.
+    * `format` (Optional) - The output format of the generated private key
+      credential. The private key will be returned from the API in PEM encoding. Options
+      include: `pkcs8`.
+
+  * `client_certificate`
+    * `common_name_template` (Optional) - A [username template](/vault/docs/concepts/username-templating)
+      to be used for the client certificate common name.
+    * `ca_cert` (Optional) - The PEM-encoded CA certificate.
+    * `ca_private_key` (Optional) - The PEM-encoded private key for the given `ca_cert`.
+    * `key_type` (Optional) - Specifies the desired key type. Options include:
```

Comment:
wondering if we should change this to be required? The `credential_config` is Optional, but if the `credential_type` is `client_certificate`, then should we let users know that this will be required for that cred type?
```suggestion
    * `key_type` (Required) - Specifies the desired key type. Options include:
```

Reply:
Yeah, I agree. It makes more sense to let the user know in case they use client cert for authentication. 👍